src/api/flaskr/service/common/storage.py

```
@@ -0,0 +1,201 @@
+@dataclass(frozen=True)
+class StorageUploadResult:
+    provider: str
+    url: str
+    bucket: str
+    object_key: str
+
+
+def _normalize_profile(profile: str) -> str:
+    resolved = (profile or "").strip().lower() or OSS_PROFILE_DEFAULT
+    if resolved not in _ALLOWED_PROFILES:
+        raise ValueError(f"Unknown storage profile: {profile}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)